### DISCUSSION TITLE
Ability to include canvas files in the search / embedding

### DISCUSSION THREAD

**Discussion by abhuva** [brianpetro/obsidian-smart-connections#116] [Ideas]
I am not entirely sure if this is possible in an easy way - but i just realized that *.canvas files dont get included in the search.
I can imagine there has to be a lot of post-processing to make them viable (by filtering out only the actual text content) - but atleast for me i use canvas quite often to sketch ideas, present certain concepts etc...

I didnt wanted to make a feature request, as i am not sure if this is really something worth looking into. 

Just to give an example: i use a plaintext accounting software (beancount, ledger) and use a canvas file to note down how my accounts are structured, what content goes where etc...
Now i thought i let summarize gpt this note - and it seems, whatever i try, its not part of the search result. If it where, it might have been possible to get interesting information out of it (together with my other normal notes on plain-text accounting).

**Comment by brianpetro** [comment]
Hey @abhuva 

I've already looked into this, and I'm excited about the prospect of supporting canvases. There's a cool side-effect: _generating new canvases with AI._

Please feel free to start a feature request. This way, especially if more people voice their desire for the feature, I can prioritize it higher and might get to it a bit sooner.

Brian🌴

**Comment by AMaestroG** [comment]
Me parece que eres el que mejor ha entendido la IA en el mundo, me apasiona hacer canvas con IA que genere 6 nuevas ramas con caminos diferentes, que genere 6 ramas del resumen de una nota o resumen WEB / PDF, o que [[genere 6 nuevos nodos]]. 6 nodos es por la [[teoría de los 6 grados de separación]] que tome el número de nodos de la variable n de la cabecera yaml por defecto 6, y jugar con variables Yaml para ajustar los resultados

**Reply by brianpetro** [reply]
Yes, that's definitely one way to go about it, generating new nodes from a starting point. 

I also think that generating nodes as connections between existing notes could be really cool. And I would love to see something like reverse-engineering a plan to a desired outcome, with generated nodes representing steps along the path. 

Thanks for the feedback on this!
Brian 🌴

**Reply by AMaestroG** [reply]
Me asombras, si consigues eso te pido matrimonio.
